Plumbing [Not dead, I'm just burnt out] [nevermind, definitely dead]

Xhuis opened the pull request:
:cl: Xhuis
add: Sinks, toilets, showers, and other water-reliant objects now draw from a plumbing tank that generates reagents for the entire station. These reagents can be changed by anyone who can access the tank.
/:cl:

WIP. I put this out because I didn't want to finish the entire feature only to have it be closed, and it's actually somewhat functional as of now.

Currently the sprite is a placeholder. If you feel like making a cool sprite I'd love to go with it.

Adds a new machine, the master plumbing tank. ___I haven't decided which job will get this yet, but I'm leaning towards engineering or atmos. I don't want to give it to janitors, because that is a job that should, in my opinion, stay shallow and not require much work.___ The tank automatically generates reagents up to a maximum of 10,000 units based on an internal "buffer" container. This is done at a rate of 1u a tick but will likely be increased if its parts are upgraded. By default, the buffer is a 30u bottle of water.

The tank has three modes: Idle, Synthesize, and Drain. Idle does just that and nothing more. Synthesize uses power to create more reagents, and Drain uses power to destroy them. Modes and all information can be read from the display, which, being a giant piece of plumbing, is difficult to understand, although there'll be a paper with helpful* notes next to it once I decide where to slap it down.

Also refactors all water-based machinery and structures to, unless overriden, draw from the master plumbing tank through bluespace magic. It isn't picky as to what it grabs, meaning that you can fill the pipes with acid and the next person to wash their hands will either lose their gloves or have their skin melt off.

I made this because it's something people have wanted for a long time and it's a fun addition to the game. I also operate under the assumption that I'll eventually be giving this to Engineering, so it gives them something to do other than shut off random APCs for amusement and micromanage the supermatter.

Also adds several helper procs for reagent holders to get the proportions of every reagent and to adjust them based on provided volume.

Commit:
First iteration of plumbing

optimumtact commented:
This isn't very mechanically interesting, can you make the machinery more complex or interesting for engineers/atmos to use? I.e require plasma or freon or similar

Xhuis commented:
Having it require gases is just tedious and has no depth to it (unless you hook it up to a gas line, of course), but I can easily see about adding different things, either in the form of required maintenance or just better output if you fine-tune it enough. Things like that.

optimumtact commented:
The supermatter does things right, every other machine is pretty much garbage.

OtherDalfite commented:
This is a cool idea, but I don't really think it offers a lot to the game. If people wanna fuck over someone using acid they'll just use a spray bottle, and nobody really washes there hands anyways. I can only see this being used to like screw over botany or medbay, not much else.

Xhuis commented:
@OtherDalfite Well, excuse me! I wash my hands in-game all the time.

Joking aside, it's not the most deep of features, but it's a little niche that people have wanted for a long time, and it's something I've always wanted to do. It can also be used in good and creative ways - like, if fires break out, engineers can make some kelotane and pump it into the pipes so that people can drink from the taps and heal their burn wounds. Overall, it's not extraordinarily deep, but it has a lot of uses, and not all of them have to be bad ones.

Niknakflak commented:
Give it to the Janitor. Engis/atmos has enough toys.  If you seriously give plumbing to anyone but the fucking janitor, what the fuck are you doing?

JStheguy commented:
No, you are all stupid.
Give it to the bartender, obviously.

Xhuis commented:
@Niknakflak While the janitor is the very obvious logical choice, it has its appeal to most people (myself included) because it's a low-stress job that can still contribute to the station. I don't want to force a job to do something when it's renowned for being laid back.

Commit:
Makes sinks function under the new system

Xhuis commented:
Sinks now work under the new plumbing system. Code is a little spaghetti because of subtypes that don't use the new system, like puddles, but it's tested and seems to work okay.

Jordie0608 commented:
@Xhuis Asis this sounds like it could be comformtably ignored by a Janitor if he didn't want to get involved with it so I don't think it'd jeopardize a janitor's laid-back approach.

Shapsy commented:
Do janitors work on plumbing? Sounds more mechanical than washing floors

Xhuis commented:
Still not sure about that part.

bgobandit commented:
HAPPENING

PKPenguin321 commented:
j a n i t o r
then give him a red hat and overalls as an alt outfit
PLEASE

Niknakflak commented:
Janitors do plumbing
Plumbers do plumbing
Mechanical engineers do not do plumbing
Atmospheric engineers do not do plumbing

GunHog commented:
I second the plumber = janitor idea, strictly because the number of players naming themselves "Mario" will increase ten-fold! On a serious note, it should not hurt the janitor's lazy, laid back lifestyle. He does not have to touch the machine unless someone breaks in and pours acid/fire/booze into it.

Poojawa commented:
Atmosia is the city of pipes. They handle all aspects of the station's environment, including water/sewage handling in addition to air. 

Also Atmosia has enough space once that offset pipes PR is put in.

Iamgoofball commented:
give it to the janitor

optimumtact commented:
@Poojawa that would only make sense if this actually had any pipes associated with it.

Poojawa commented:
Wasn't it assumed already that the pipes were already linked to all the sinks/toliets, but they would otherwise be too much of a pain to have to manage from a coding aspect?

Xhuis commented:
Yeah, currently there are no plumbing pipes and it operates through code magic. I'll come up with fluff for it but I don't want to use pipes for it. I will be adding construction for sinks and whatnot that requires pipes, but they won't be linked physically to the tank.

There's still some division over the role that should get the tank, so I made a poll. Place ya bets.

Xhuis commented:
Fucking phone.
http://www.strawpoll.me/12575071

ShadowDeath6 commented:
wasn't there another "plumbing" pr that got disliked because it was infinite duplication of meme chems?
I can't find it but I swear there was one

Supermichael777 commented:
clearly this is a job for the Chaplin

KorPhaeron commented:
It doesn't matter what job you give it to unless you give it to chemistry because they are the only one with access to liquids that would meaningfully impact this.

Other than yelling at botany to bring one of their 10k unit tanks so they can fill the sinks roundstart I guess

RemieRichards commented on code/modules/plumbing/_the_tank.dm:
```diff
@@ -0,0 +1,132 @@
+				reagents.remove_reagent(R.id, 1)
+
+/obj/machinery/plumbing_tank/interact(mob/user)
+	var/dat
```
```var/list/dat = list()```

Xhuis replied:
Not once have I ever seen dat as a list. Why use it now?

RemieRichards replied:
>not once

Wow, well you've clearly never looked at a string UI in a while have you?
>Why use it now?

Relatively new feature (past year or so) is why. ```/list/Join()``` turns a list of strings into one string, simple enough.
Say I have this string: "C"
Then I ```+= "A"``` I now have "CA"
Then I ```+= "T"```` I now have "CAT"

The overall difference is "C" -> "CAT", Ok, makes sense.
But what has the string table done in the background?
NEW STRING "C"
NEW STRING "CA"
NEW STRING "CAT"
These strings will sit in the table, for a decent amount of time, in the hopes that they'll be used again, the issue is, NOBODY IS GOING TO USE "CA", "CA" SHOULD NOT BE IN THE TABLE, the *perfect* outcome would be:
NEW STRING "CAT"
("NEW STRING" here means "look through every single string and see if one of them matches, if so reuse it, else find an open spot and add the new one" spoilers: this isn't cheap when you're doing it every other line)

and by using a list, we get N strings (where N is the number of ```+=```'s) + 1 for the final string, now this *isn't* the perfect case, but it's a shit load better.

```/list/Join()``` was requested specifically for these kinds of situations, other languages with string tables/trees include a ```StringBuilder``` class, that holds onto each of the component strings, and produces one final string when you ```StringBuilder.finalize()```, our list + ```.Join()``` fulfils this role in DM.

tl;dr performance you dingbat

GunHog replied:
Wow, I did not know of this either, lol!

xxalpha replied:
This is basic beyond mechanics, guys. It's like you don't go to coderbus.

Xhuis replied:
I'm in there every day. I just added a console that used the old method and nobody batted an eye. Forgive my skepticism.

GunHog replied:
Maybe I was not in the chat at the time this standard was made, same with Xhuis! Most of us which missed that discussion will only find out when RR comes to correct us!

RemieRichards replied:
There was no discussion really, but I've enforced this for quite literally as long as the feature has existed (year, year+)

xxalpha replied:
It's been known for a good while now, but I am just messing with ya.

RemieRichards commented on code/modules/plumbing/_the_tank.dm:
```diff
@@ -0,0 +1,132 @@
+	dat += "<br><br>"
+	dat += "<a href='?src=\ref[src];switch_modes=1'>Set the OPMD dial to a new setting</a>"
+	var/datum/browser/popup = new(user, "plumbing_tank", name, 600, 400)
+	popup.set_content(dat)
```
```dat.Join()```

RemieRichards commented on code/modules/plumbing/_the_tank.dm:
```diff
@@ -0,0 +1,132 @@
+		switch(new_mode)
+			if("IDLE")
+				operating_mode = PLUMBING_TANK_IDLE
+			if("SYNT")
```
just add an extra letter, it makes SYNTH and DRAIN correct, and IDLE already is.

Xhuis replied:
They're restricted to four letters for fluff purposes. The display is meant to be obscure.

RemieRichards commented on code/modules/plumbing/sink.dm:
```diff
@@ -0,0 +1,181 @@
+		return
+	if(!Adjacent(user))
+		return
+	if(!reagents.total_volume && !plumbing_has_reagents(15) && use_reagents) //15 units per use
```
put ```use_reagents``` first.

RemieRichards commented on code/modules/plumbing/sink.dm:
```diff
@@ -0,0 +1,181 @@
+		washing_face = 1
+	user.visible_message("<span class='notice'>[user] starts washing their [washing_face ? "face" : "hands"]...</span>", \
+						"<span class='notice'>You start washing your [washing_face ? "face" : "hands"]...</span>")
+	busy = 1
```
proper ```TRUE/FALSE```, come on.

Xhuis replied:
This is legacy code that was moved to a new file. It'll be touched up after I migrate the other objects.

RemieRichards commented on code/modules/plumbing/sink.dm:
```diff
@@ -0,0 +1,181 @@
+
+	busy = 0
+
+	if(!reagents.total_volume && !plumbing_has_reagents(15) && use_reagents)
```
```use_reagents``` first.

RemieRichards commented on code/modules/plumbing/sink.dm:
```diff
@@ -0,0 +1,181 @@
+
+	if(!istype(O))
+		return
+	if(O.flags & ABSTRACT) //Abstract items like grabs won't wash. No-drop items will though because it's still technically an item in your hand.
```
combine these two ifs.

RemieRichards commented on code/modules/plumbing/sink.dm:
```diff
@@ -0,0 +1,181 @@
+		reagents.clear_reagents()
+		user.visible_message("<span class='notice'>[user] washes [O] using [src].</span>", \
+							"<span class='notice'>You wash [O] using [src].</span>")
+		return 1
```
```TRUE```

RemieRichards commented on code/modules/reagents/chemistry/holder.dm:
```diff
@@ -743,6 +743,24 @@ var/const/INJECT = 5 //injection
+/datum/reagents/proc/get_reagent_proportions(include_empty_space = FALSE) //Returns an assoclist with all reagents equal to the percentage of the container they're taking up
+	var/list/reagent_props = list()
+	if(!include_empty_space)
+		for(var/datum/reagent/reagent in reagent_list)
```
istypeless please.

Kraseo commented:
now i can finally fluoridate the station's water supply

optimumtact commented:
FLOURIDATION CAUSES CANCER SCREEE

factoryman942 commented:
>be cmo
>draw 5u omnizine from your hypospray into  a bottle
>infinite omnizine

optimumtact commented:
also yeah this is basically a massive duplication system for chems, guess I'm 👎  on it

Kraseo commented:
omnizine is trash

RemieRichards commented:
could make the machine use a godawful amount of power when set to anything not water.

FantasticFwoosh commented:
@KorPhaeron giving to chemistry is a terrible idea because they or somebody in medical would pollute the water system readily every round with flourine or some other dangerous meme chemical (virologist drops entire bottle of infection diseases like legionnaires down the drain) 

> Right now the possibilities are scary, especially with xenobio chemicals, you'd have to constantly test the taps.

[I already had my own suggestion on who you could give it to,](https://tgstation13.org/phpBB/viewtopic.php?f=9&t=10257&p=271016#p270233) given that engineering should focus on the supermatter not going nuclear on them/maintenance and atmos usually steps in to help engineering and has their own semi complicated air system to handle. Besides the meme content of my suggestion its all very rational

> Emagging the water machine to produce sickness/toxicity and plant harming sewage

Xhuis commented:
The potential for reagent duplication is somewhat circumvented by the staggering amount of water already in the system meaning that you have to spend ten minutes draining the entire system. The Odysseus does the same thing already and at a much faster rate. That said, I'm going to blacklist certain reagents and will add some sort of cost to duplicating non-whitelisted reagents.

FantasticFwoosh commented:
We have biohazard disposal i guess for just emergency dumping water, won't water dissapear when it touches space?

Xhuis commented:
I don't know his real-life physics dictate what happens in that case, but for the purposes of this PR it can be arranged.

kevinz000 commented:
>i can't autism with pumping pipes

*glares*
did you implement liquid spills yet because I want to fill a room with water and see who tries going into it :^)

FantasticFwoosh commented:
@PKPenguin321 give the jantior a green outfit and the plumber a red one.

Put the second job slot to have the spare yellow & purple outfits

KorPhaeron commented:
>The potential for reagent duplication is somewhat circumvented

Isn't that the sole feature of this though?

I can see this being used for:

1) Healing chems for the whole station

2) People making holocaust jokes in the showers

Both this system and our chemicals in general are pretty limited so there doesn't seem to be much room to do anything else

Xhuis commented:
...I can't think of an argument against that. Yeah, you have a point.

Xhuis commented:
In other news, @Nienhaus has graciously provided me sprites, so I'll be using those and posting a picture soon.

Anonmare commented:
In regards to who should get the tank. I say put it near atmos and give both janitors and atmostechs access to it. Should be a decent compromise, no?

Also I hope this will allow for the potential of slippery-sloping into implementing liquid flooding one day, we are still playing an overdue signed atmospherics simulator after all. That and I like the idea of locking someone in the bathroom while it floods.

Iatots commented:
Really not a fan of another magic bluespace feature, though even a real implementation with piping would be a nightmare with assistants unwrenching water pipes to flood the entire station.

Jalleo commented:
@optimumtact people will probably abuse for two weeks realise its weak as fuck and then almost never use it.

TrustyGun commented:
i dont see why you shouldnt give it to the janitor, its not like this feature is super complicated and crazy

AdamElTablawy commented:
If you want it to have more impact than a simple niche thing, you'll want to add piping rather than bluespess magic, tie a few machines to it  and probably add water itself flowing and flooding.

For example, cryogenics AND the cloning pods both seem to have the occupant immersed in some kind of fluid, and I doubt that a tiny beaker can cover an entire human and more as the cryo tank seems to indicate. So, cryogenics fluid uses plumbing for water. Sabotage cryo? You don't just need to modify the beaker-You can dick with plumbing instead and suddenly cryo's fluid is fucking acid.

Cyberboss commented:
With the planetstation stuff upcoming, design decisions about water have to be made eventually

Xhuis commented:
I've been chatting with @AdamElTablawy and I'm aiming to expand this system a bunch in the future. Adding stuff like sprinklers to put out minor fires, adding liquid cooling to certain machines, that kind of thing.

kevinz000 commented:
On the note of pipe closers our pipes don't even leak if there's an opening so we don't need it :P

Bawhoppen commented:
What is the point in plumbing without actual pipes???

Xhuis commented:
What's the point of teleportation if you don't need to link them with wires?

I'd like pipes but it'd require a tremendous amount of work from mappers and players. I don't want _four_ stationwide lines.

Niknakflak commented on code/modules/plumbing/sink.dm:
```diff
@@ -0,0 +1,181 @@
+			H.lip_color = initial(H.lip_color)
+			H.wash_cream()
+			H.regenerate_icons()
+		user.drowsyness = max(user.drowsyness - rand(2,3), 0) //Washing your face wakes you up if you're falling asleep
```
woah, EMERGENT GAMEPLAY

Xhuis replied:
M E T A
E
T
A

Iatots replied:
showers should totally have this, especially when freezing.

Niknakflak commented on code/modules/plumbing/sink.dm:
```diff
@@ -0,0 +1,181 @@
+	icon_state = "sink_alt"
+
+
+/obj/structure/sink/puddle	//splishy splashy ^_^
```
comment not required but definitely appreciated

Xhuis replied:
This came from legacy code, I'm not _that_ autistic

Militaires replied:
splishy splashy

Xhuis commented:
DM is being finnicky and sometimes randomly never initializes the lighting subsystem. Testing on this probably won't happen quickly.

Commit:
Big patch

Xhuis commented:
* Added damage, destruction, and repairing.
  * From its useless state, add five metal, weld, and wrench. Done!
* Added the ability to eject and add buffers.
* Code cleanup in response to maintainer review.

FantasticFwoosh commented:
@KorPhaeron the problem with this is that you're forgetting its a WATER system and all the nooks and crannies it delivers WATER to the station else they use the water tanks (actually remove water tanks or use them as water canisters locally would be easier)

> There will be literal napalm showers, and phosphorus exploding sinks/pipes that punish people pouring water down a drain. You cant water plants with healing chems, the most annoying thing may be holy water for ruining a cultists day.

> Create canister pad (or water variant) with attached water pipes, to divert water in your 'gas form' to fuel the taps for current water/fuel containers meaning you can have all of botany's water supply in taps in the back room or maintenance without wiring it into the grid

> Also how will this new gas system affect Toxins? I have a suggestive thread up.

AdamElTablawy commented:
The issue with plumbing is-

Bombs become more unrepairable.

Water comes gushing out of broken pipes, so one busted pipe instantly dicks the water system and also makes breaches slip hell, which while amusing is still irritating.

That's 4 maps for plumbin'.

There should be drains, pumps, etc, but that's both out of the scope of this pr probs and also shouldn't be interconnected in that way.

FantasticFwoosh commented:
A small localised pipe system linked to tanks in places such as starboard storage & maint could work for closed water supplies not attached to the mains. As i previously mentioned.

A bit like bringing a air canister with you to go with the aux base, putting in a input then tying up the output to connect to the 'disposal water' network to be shot out of biohazard disposal into space where it gets sucked into the void.

Xhuis commented:
I should elaborate that the scope of this PR is mainly to migrate all existing water-based structures to the new system. That's the extent of it currently - other features will come in later PRs after people have had a chance to demo the system and if they like it.

Commit:
Makes showers function with plumbing

RemieRichards commented on code/modules/plumbing/shower.dm:
```diff
@@ -0,0 +1,223 @@
+/obj/machinery/shower
```
good job, thanks for making the diff hard to read.

Xhuis replied:
Sorry, I guess? The parts I changed are everything involving reagents. I'll go through and highlight the particulars in a few but something just came up so it'll take me ~20 minutes at the most, hopefully

Xhuis commented:
Showers are done, but keep in mind that a lot of is legacy code moved to a new file so it might not be up to scratch.

Xhuis commented:
![remietime](https://cloud.githubusercontent.com/assets/15898043/24272402/940da98c-0ff3-11e7-9e4d-30fd44d3caaf.png)

RemieRichards commented:
you had that prepared!

Xhuis commented:
I totally did.

Cyberboss commented:
I am saving this pic forever omfg

On Mar 23, 2017 6:09 PM, "Xhuis" <notifications@github.com> wrote:

> I totally did.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/25337#issuecomment-288875820>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHywesDkxjDU2a2qunkaGWPxKR0xNhFaks5rou2KgaJpZM4MjNP6>
> .
>

Xhuis commented:
Turns out 20 minutes was a vast underestimate. Give me a minute...

Xhuis commented on code/modules/plumbing/shower.dm:
```diff
@@ -0,0 +1,223 @@
+	var/watertemp = "normal"	//freezing, normal, or boiling
+
+/obj/machinery/shower/Initialize()
+	create_reagents(5)
```
This is new, obviously.

Xhuis commented on code/modules/plumbing/shower.dm:
```diff
@@ -0,0 +1,223 @@
+
+
+/obj/machinery/shower/attack_hand(mob/M)
+	if(!on && !plumbing_has_reagents(5) && !reagents.total_volume)
```
Showers can't function without water now.

Xhuis commented on code/modules/plumbing/shower.dm:
```diff
@@ -0,0 +1,223 @@
+	master_plumber.request_liquid(src, 5)
+	if(on)
+		process()
+
```
I also made this do what process() does because they do the same thing.

Xhuis commented on code/modules/plumbing/shower.dm:
```diff
@@ -0,0 +1,223 @@
+
+/obj/machinery/shower/proc/wash_mob(mob/living/L)
+	L.wash_cream()
+	if(iscarbon(L))
```
I cut the fire stack removal out of here, as I think it's what water does anyhow.

Xhuis commented on code/modules/plumbing/shower.dm:
```diff
@@ -0,0 +1,223 @@
+		reagents.reaction(loc, TOUCH)
+		for(var/atom/movable/G in loc)
+			reagents.reaction(G, TOUCH)
+			if(isliving(G))
```
Added the reagents reaction here. At the end of the line it clears all of them.

Xhuis commented:
That should be all. The reagents are also heated or cooled to match if the water temperature is set to something.

coiax commented:
@Xhuis I feel that you should probably remove the duplication aspect, and just start with a large water tank hooked up. I don't really like getting something "for nothing" as it were. I've got my eyes on cryo for that reason as well.

FantasticFwoosh commented:
Its water for energy, else you have to do the annoying ordering water from cargo thing like we currently have to just buy more oxygen on the station in the rare instances we need it or atmos pipes are screwed.

If you want take up "Dirty deeds done dirt cheap" reasoning, go confront mekhi about his gas mining operations that magically make gas to replace virtually all the map gas cylinders. 

> (also destroy the water tank and enough water to flood the entire station 50 times comes rushing out and insta kills everything)

coiax commented:
Ordering more water from cargo is exactly the sort of shit we should be encouraging.

Xhuis commented:
once this is on the maps it should be ready for what this PR is aiming for. balance issues aside.

kevinz000 commented:
Honestly for the chemical stuff it should be poured in and only water and some basic reagents should be synthesized. Would be quite OP if some certain reagents got in there.

Xhuis commented:
I could see that working but at that point you'd require massive amounts of reagents. Which I suppose would be an effective balancer, in all honestly.

KorPhaeron commented:
@tgstation/commit-access 

This has been open 9 days now. If someone doesn't come along and declare they're willing to help this be finished and then hit the merge button within the next 48 hours I'm going to assume no maintainers want it and close it

Xhuis commented:
You and 48 hour ultimatums, Kor. I don't think that that link even pings anyone as it leads to a 404 page. 

As for finishing this, its initial content is passable. I've just been burnt out lately and haven't been actively working on it, and I've been playing other games that don't sap my creative energies so much. It still needs balancing and putting on the map, so if I'm not up for working on it tomorrow I'll close it myself and reopen it once I've done work on it.

RemieRichards commented:
@Xhuis it's a 404 for you, it's the list of contributors with commit access to those who have commit access (read: again, not you)

Iamgoofball commented:
WHOA

LEAKING MAINTAINER SECRETS

RemieRichards commented:
nothing about that is secret, you can work it all out from other sources.
oranges, Raz, MSO, Serverbot and Dorsis all have commit access but have their settings on "Private" in the team, so they don't show up, but everyone knows Dorsis tags things, Raz is a maintainer, oranges is, MSO has access because it's mso and the bot has it because it needs it for changelogs.

and the huge moby dick copypasta is... moby dick, a public book.

tl;dr shove off goofballs.

Iamgoofball commented:
L
E
A
K
I
N
G

AnturK commented:
It's okay it's not our world domination org projects.

As for the feature itself, i wouldn't mind it if it wasn't so effortless. I feel like it's healing sinks is going to become another "ugprade orm" must-have that happens every round because it's laughably easy.  And medbay made obsolete is  not something i'd like to see every round.

RemieRichards commented:
@AnturK yeah, those are on the repositories tab, which isn't selected in my screenshot.

AnturK commented:
They'll never know.

optimumtact commented:
mine is usually public but I keep getting removed from the list

lzimann commented:
@optimumtact Have you tried not commiting treason over and over again?

coiax commented:
I don't want any synthesis of new reagents, I just want a giant tank that needs to be refilled.

RemieRichards commented:
or that.

lzimann commented:
About the PR: I don't think a giant tank that magically fills every sink with the reagent you put inside is fun in any way, it's basically no improvement from what it is today.

GunHog commented:
>About the PR: I don't think a giant tank that magically fills every sink with the reagent you put inside is fun in any way, it's basically no improvement from what it is today.

Pouring clf3 into it to create FIRESHOWERS sounds like fun to me! Anyway, not having any sort of reagent creation or multiplication could mean that anything the janitor adds would affect at most only a few people. 

How else is anyone lacking access to a chem machine going to produce enough reagents to meaningfully affect people?

Xhuis commented:
A few thoughts:

* A tank that has to be refilled can work, but it doesn't offer anything new besides a chore for someone to do. It's still possible to do though, and very easy.
* The whole point of reagent synthesis is to allow room for creativity. People might fill the sinks with burn medicine in the event of a fire to keep people alive, for instance. I think it _is_ an improvement over what we have, but it's admittedly a shallow one.
* If the feature really is "effortless", I'd be happy to make changes to it and try to make it more fun and interesting for everyone involved. While this is the framework, I'm still intending to add more at some point. At least, I would, if I wasn't burned out.

I'll have a verdict on whether or not I'll close this to continue later by this evening. Leaning towards that way though.

KorPhaeron commented:
@gun_hog

It is popular because it sounds cool on the surface (fire showers!)

But in reality it just makes medbay even more redundant and worthless when you dump a beaker of omnizine or whatever in roundstart.

And it isn't worth throwing an already suffering department under the bus for the sake of fire showers.

Ironically, several of the people who upvoted this also post on the forums about wanting deeper medical.

Xhuis commented:
One of its outstanding issues currently is that it lacks any balancing at all. You can just shove anything in there and get infinite amounts of it. Before the PR is merged I'd like to change the balancing in it. I'll probably make a whitelist of reagents that are cheap to produce, with the rest being slow and heavy on power drain. If some idiot tries to fill the pipes with deadly toxins, the power drain will be evident enough that an engineer can just shut off power remotely.

There are also other ways, and I'm open for suggestions.

coiax commented:
@KorPhaeron I don't want to get too pedantic, but omnizine is not a chemical availible at roundstart. The best you could do with a generic roundstart cureall is tricoord.

Xhuis commented:
Iirc you can grind warm donk pockets for omnizine. I used to do it for strange reagent roundstart as a chemist.

coiax commented:
Oh

KorPhaeron commented:
I think someone put adminrozine in the mining pens as well so if you're willing to do the grinding step you could have a magic nanite dispenser in every department within the first 5-10 minutes of the round

coiax commented:
I thought they took the nanites out of the mining pens.

Anyway, I don't like magical synthesis of reagents in general, so I vote for giant tank of water.

Xhuis commented:
I can't muster up the energy to work on this right now, and I'm not going to let it stay on the tracker for ages. I'll close this for now, and once I've made some progress open it up again.

GunHog commented:
Good luck Xhuis! We are counting on you not to let this wind up abandoned!

ShizCalev commented:
Suggestion; Give emagged chem dispensers have an option to dump contents of a beaker into the water line (since they distribute water, one would assume they were already connected to it)

Xhuis commented:
@GunHog I didn't leave you hanging with shadowling, even if it did end up being shut down. I'm hoping this one won't go bad.

@ShizCalev Seems very strong to me but I may consider it.

ShizCalev commented:
Will shuttles have to have a water tank added since some of them have showers/sinks too?

Xhuis commented:
Probably, yeah.
